### DISCUSSION TITLE
Error trying to create table from javascript

### DISCUSSION THREAD

**Discussion by dsgeyer** [supabase/supabase#28725] [Questions]
Loom: https://www.loom.com/share/25a93f3ac4fb4dba9d0d14cda7a53d9f?sid=e7405aa0-0a2e-4d17-bc72-3a8827f35161

I created a function in my database to create a table. 

I try to call that from my javascript code and I get this error: 

Error creating documents table: {
  code: '42501',
  details: null,
  hint: null,
  message: 'permission denied for schema public'
}

My supabase function called "ccxm_create_documents_table"
"""
begin
    create table test_table3 (
        id bigint primary key generated always as identity,
        content text,
        metadata jsonb,
        embedding vector(3072)
    );
end;
"""

Javascript:
"""import { createClient } from "@supabase/supabase-js";
import dotenv from "dotenv";
dotenv.config();

// Initialize Supabase client
const sbClient = createClient(
  process.env.MY_SUPABASE_URL,
  process.env.MY_SUPABASE_KEY,
  {
    db: { schema: "public" },
  }
);

// Function to create the documents table
async function createDocumentsTable() {
  try {
    const { data, error } = await sbClient.rpc("ccxm_create_documents_table");

    if (error) {
      console.error("Error creating documents table:", error);
    } else {
      console.log("Documents table created successfully:", data);
    }
  } catch (error) {
    console.error("Unexpected error:", error);
  }
}

// Main function to run the script
async function main() {
  await createDocumentsTable();
}

// Run the main function
main().catch(console.error);"""

**Comment by GaryAustin1** [comment]
The REST API roles only have usage permission on the public schema.  You would need to make your function security definer type 
and you really need to limit the role grants on the function or anyone could create tables (assuming you are adding the ability to create more than one table as your plan).

In general though it is NOT a good idea to create tables with functions.  They should be part of your planned schema and have RLS on them.
